src/test/java/com/wikia/webdriver/common/driverprovider/DriverProvider.java

```diff
@@ -0,0 +1,59 @@
+package com.wikia.webdriver.common.driverprovider;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.wikia.webdriver.common.core.WikiaWebDriver;
+import com.wikia.webdriver.common.core.configuration.Configuration;
+import com.wikia.webdriver.common.core.drivers.Browser;
+
+public class DriverProvider {
+
+  private static final List<WikiaWebDriver> drivers = new ArrayList<>();
+  private static int ACTIVE_BROWSER_INDEX = 0;
+
+  private DriverProvider() {
+  }
+
+  private static void newInstance() {
+    drivers.add(Browser.lookup(Configuration.getBrowser()).getInstance());
+  }
+
+  private static WikiaWebDriver getBrowserDriver(int index) {
+    for (; drivers.size() <= index; ) {
+      newInstance();
+    }
+
+    return drivers.get(index);
+  }
+
+  private static WikiaWebDriver getBrowserDriver() {
+    return getBrowserDriver(0);
+  }
+
+  public static WikiaWebDriver getActiveDriver(){
+    return getBrowserDriver(ACTIVE_BROWSER_INDEX);
+  }
+
+  public static WikiaWebDriver switchActiveWindow(int index){
+    ACTIVE_BROWSER_INDEX = index;
+    return getActiveDriver();
+  }
+
+  public static void close() {
+    try {
+      for (WikiaWebDriver webDriver : drivers) {
+        if (webDriver.getProxy() != null){
+          webDriver.getProxy().stop();
+        }
+        if (webDriver != null) {
+          webDriver.quit();
+        }
+      }
+    }catch (Error e){
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either remove or fill this block of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS00108)
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Catch Exception instead of Error. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1181)